### DISCUSSION TITLE
可以通过图片的md5(uuid)值获取图片的URL吗

### DISCUSSION THREAD

**Discussion by Under-estimate** [mamoe/mirai#1005] [Q&A] [answered]
下面是我在群聊中收到的一条图片分享xml消息
```xml
<?xml version='1.0' encoding='UTF-8' standalone='yes' ?>
<msg serviceID="5" templateID="12345" action="" brief="0qx姐姐的照片" sourceMsgId="0" url="" flag="0" adverSign="0" multiMsgFlag="0">
    <item layout="0">
        <image uuid="B8549F7AA6064A7B1321EC3B40B6138F.jpg" md5="B8549F7AA6064A7B1321EC3B40B6138F" GroupFiledid="0" filesize="81322" local_path="B8549F7AA6064A7B1321EC3B40B6138F.jpg" minWidth="200" minHeight="200" maxWidth="500" maxHeight="1000" />
    </item>
    <source name="樱?(id:73388316 author:JYT)" icon="" action="" appid="-1" />
</msg>
```
其中只包含有图片的md5(uuid)值，并没有图片的URL地址或者图片id，请问这种情况可以获取到图片的URL地址吗？

**Comment by Him188** [comment]
目前没有这样的API，但可以自行构造 Image 然后查询链接

**Comment by andogy** [comment, marked as answer]
> 下面是我在群聊中收到的一条图片分享xml消息
> 
> ```
> <?xml version='1.0' encoding='UTF-8' standalone='yes' ?>
> <msg serviceID="5" templateID="12345" action="" brief="0qx姐姐的照片" sourceMsgId="0" url="" flag="0" adverSign="0" multiMsgFlag="0">
>     <item layout="0">
>         <image uuid="B8549F7AA6064A7B1321EC3B40B6138F.jpg" md5="B8549F7AA6064A7B1321EC3B40B6138F" GroupFiledid="0" filesize="81322" local_path="B8549F7AA6064A7B1321EC3B40B6138F.jpg" minWidth="200" minHeight="200" maxWidth="500" maxHeight="1000" />
>     </item>
>     <source name="樱?(id:73388316 author:JYT)" icon="" action="" appid="-1" />
> </msg>
> ```
> 
> 其中只包含有图片的md5(uuid)值，并没有图片的URL地址或者图片id，请问这种情况可以获取到图片的URL地址吗？

如果是在qq接收到的话
https://gchat.qpic.cn/gchatpic_new/0/0-0-(这里填md5)/0
